ui/component/livestreamChatLayout/view.jsx

```diff
@@ -0,0 +1,375 @@
+  const {
+    claim,
+    comments: commentsByChronologicalOrder,
+    embed,
+    isPopoutWindow,
+    pinnedComments,
+    superChats: superChatsByAmount,
+    uri,
+    doCommentList,
+    doCommentSocketConnect,
+    doCommentSocketDisconnect,
+    doResolveUris,
+    doSuperChatList,
+  } = props;
+
+  const {
+    location: { pathname },
+  } = useHistory();
+
+  const isMobile = useIsMobile();
+
+  const discussionElement = document.querySelector('.livestream__comments');
+
+  const restoreScrollPos = React.useCallback(() => {
+    if (discussionElement) discussionElement.scrollTop = 0;
+  }, [discussionElement]);
+
+  const commentsRef = React.createRef();
+
+  const [viewMode, setViewMode] = React.useState(VIEW_MODES.CHAT);
+  const [scrollPos, setScrollPos] = React.useState(0);
+  const [showPinned, setShowPinned] = React.useState(true);
+  const [resolvingSuperChats, setResolvingSuperChats] = React.useState(false);
+  const [mention, setMention] = React.useState();
+  const [openedPopoutWindow, setPopoutWindow] = React.useState(false);
+  const [chatHidden, setChatHidden] = React.useState(false);
+
+  const quickMention =
+    mention && formatLbryUrlForWeb(mention).substring(1, formatLbryUrlForWeb(mention).indexOf(':') + 3);
+  const claimId = claim && claim.claim_id;
+  const commentsToDisplay = viewMode === VIEW_MODES.CHAT ? commentsByChronologicalOrder : superChatsByAmount;
+  const commentsLength = commentsToDisplay && commentsToDisplay.length;
+  const pinnedComment = pinnedComments.length > 0 ? pinnedComments[0] : null;
+
+  let superChatsChannelUrls = [];
+  let superChatsFiatAmount = 0;
+  let superChatsLBCAmount = 0;
+  if (superChatsByAmount) {
+    superChatsByAmount.forEach((superChat) => {
+      const { is_fiat: isFiat, support_amount: tipAmount, channel_url: uri } = superChat;
+
+      if (isFiat) {
+        superChatsFiatAmount = superChatsFiatAmount + tipAmount;
+      } else {
+        superChatsLBCAmount = superChatsLBCAmount + tipAmount;
+      }
+      superChatsChannelUrls.push(uri || '0');
+    });
+  }
+
+  function toggleSuperChat() {
+    if (superChatsChannelUrls && superChatsChannelUrls.length > 0) {
+      doResolveUris(superChatsChannelUrls);
+
+      if (superChatsByAmount.length > LARGE_SUPER_CHAT_LIST_THRESHOLD) {
+        setResolvingSuperChats(true);
+      }
+    }
+    setViewMode(VIEW_MODES.SUPERCHAT);
+  }
+
+  function handlePopout() {
+    const newWindow = window.open('/$/popout' + pathname, 'Popout Chat', 'height=700,width=400');
+
+    // Add function to newWindow when closed manually
+    // show the chat again and re-connect to websocket
+    newWindow.onbeforeunload = () => {
+      if (claimId) doCommentSocketConnect(claimId);
+      setPopoutWindow(undefined);
+    };
+
+    if (window.focus) newWindow.focus();
+    setPopoutWindow(newWindow);
+
+    // Disconnect since it will re-connect on the new window
+    if (claimId) doCommentSocketDisconnect(claimId);
```

Comment:
@tzarebczan, we use the commentron websocket to receive viewer updates on the livestream page.  If we disconnect is, it won't update, right?

```
    if (response.type === 'viewers') {
      const connected = response.data.connected;
      dispatch({
        type: ACTIONS.VIEWERS_RECEIVED,
        data: { connected, claimId },
      });
    }
```

Reply:
Yes, but it will be connected in the new pop out window. Then when that's closed, reconnects / calls comment list. 

This prevents a double count between main + pop out.

Reply:
It doesn't seem like the websocket session is shared between Popup and Main.  Once Main got disconnected, `viewers` is not received, so the viewer count is stuck.